src/proxy-ui-api/frontend/tests/e2e/specs/ss-backupandrestore.js

```diff
@@ -0,0 +1,270 @@
+      'successfully created',
+    );
+
+    // Get the backend-generated name of the backup and close snackbar
+    const createdBackupFileNameTextObject = await browser.getText(
+      'xpath',
+      mainPage.elements.snackBarMessage,
+    );
+    const createdBackupFileName = createdBackupFileNameTextObject.value.split(
+      ' ',
+    )[1];
+    console.log('Created backup: ', createdBackupFileName);
+    mainPage.closeSnackbar();
+
+    // Filtering backup list with the name of created backup there should be only one backup in the list
+    backupAndRestoreTab.enterFilterInput(createdBackupFileName);
+    browser.expect
+      // xrd-table has thead but no tbody so use '/tr' to get only the row elements that are direct children of table
+      .elements('//table[contains(@class, "xrd-table")]/tr')
+      .count.to.equal(1);
+    backupAndRestoreTab.clearFilterInput();
+
+    // Delete created backup (click cancel first time)
+    browser.waitForElementVisible(
+      `//table[contains(@class, "xrd-table")]//td[text() = "${createdBackupFileName}"]/following-sibling::td//button[contains(@data-test, "backup-delete")]`,
+    );
+    backupAndRestoreTab.clickDeleteForBackup(createdBackupFileName);
+    browser.waitForElementVisible(deleteBackupConfirmationDialog);
+    deleteBackupConfirmationDialog.cancel();
+    browser.expect.element(deleteBackupConfirmationDialog).to.not.be.visible;
+
+    await deleteBackup(browser, createdBackupFileName);
+    browser.end();
+  },
+
+  'Download and import backup': async (browser) => {
+    const mainPage = browser.page.ssMainPage();
+    const settingsTab = mainPage.section.settingsTab;
+    const backupAndRestoreTab = settingsTab.section.backupAndRestoreTab;
+    const backupFileAlreadyExistsDialog =
+      backupAndRestoreTab.section.backupFileAlreadyExistsDialog;
+    const deleteBackupConfirmationDialog =
+      backupAndRestoreTab.section.deleteBackupConfirmationDialog;
+
+    // delete existing backups from test dir
+    const testDataDir = __dirname + browser.globals.e2etest_testdata + '/';
+    const regex = /^conf_backup/;
+
+    fs.readdirSync(testDataDir)
+      .filter((file) => regex.test(file))
+      .map((file) => fs.unlinkSync(testDataDir + file));
+
+    // Navigate to backup and restore page
+    navigateToBackupAndRestorePage(browser);
+
+    // Create backup
+    backupAndRestoreTab.clickCreateBackup();
+
+    // Make sure backup was successfully created
+    browser.assert.containsText(
+      mainPage.elements.snackBarMessage,
+      'successfully created',
+    );
+
+    // Get the backend-generated name of the backup and close snackbar
+    const createdBackupFileNameTextObject = await browser.getText(
+      'xpath',
+      mainPage.elements.snackBarMessage,
+    );
+    const createdBackupFileName = createdBackupFileNameTextObject.value.split(
+      ' ',
+    )[1];
+    console.log('Created backup: ', createdBackupFileName);
+    mainPage.closeSnackbar();
+
```

Comment:
Could the backup creation be in a separate function since it is used in multiple tests? Also could the backup only be created once maybe in the first test case and then deleted in the last test case (where the backup is now created and deleted for a third time)?

Reply:
Good point, not sure why I didn't separate that block to a function. 

I'm not sure what would be best practise, but I've preferred changing the state of test environment in each test separately to the wanted state so that adjacent tests won't become dependent on the previous being successfull. In this case it might have been a better approach but when implementing a larger test set it becomes really complicated to have that kind of expected state. I think it would make tests easier to be broken by accident. So I've decided to be consistent on this.